sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/DeduplicateRelations.scala

```diff
@@ -223,6 +223,11 @@ object DeduplicateRelations extends Rule[LogicalPlan] {
         if findAliases(aggregateExpressions).size == aggregateExpressions.size =>
         Nil
 
+      case oldVersion @ Filter(condition, _)
+          if condition.references.intersect(left.outputSet).nonEmpty =>
+        val newVersion = oldVersion.copy(condition = condition)
```

Comment:
isn't `oldVersion` exactly the same with `newVersion`?

Reply:
You mean mean to write directly as Seq(oldVersion, oldVersion) ？

Reply:
It should be written directly as newVersion = oldVersion.copy() ?